src/shady-css/node-visitor.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+const path = Symbol('path');
+
+/**
+ * Class that implements a visitor pattern for ASTs produced by the Parser.
```

Comment:
Let me think on this for a second... I see the way you're doing dispatch in visit().